execution/function/functions.go

```diff
@@ -434,6 +434,22 @@ var Funcs = map[string]FunctionCall{
 			return float64(t.Year())
 		})
 	},
+	"absent": func(f FunctionArgs) promql.Sample {
+		if len(f.Points) == 0 {
+			return promql.Sample{
+				Metric: f.Labels,
+				Point: promql.Point{
+					T: f.StepTime,
+					V: 1,
+				},
+			}
+		}
+
+		return promql.Sample{
+			Metric: f.Labels,
+			Point: promql.Point{},
+		}
```

Comment:
I don't think this is the correct. `absent` checks whether the series exists ot not, not samples. When the series exists, I think what you want is to drop the whole series, not returning an empty point. Implementing a function like this couldn't achieve the goal of dropping the series right now, you have to refactor the framework to do that.

Probably a dedicated `absent` operator is easier.

Reply:
@yeya24 Thank you for the guidance. I will go through @MichaHoffmann implementation to understand it.